### PR TITLE
build: update dev image references

### PR DESCRIPTION
**Pull request by huniafatima-99**
This PR updates the dev images references in docker-compose.yml file for the following IDAs

- edx-platform (cms + lms)
- ecommerce
- registrar
- credentials
- edx-notes-api-dev

#### Issue with CLI tests failing on macOS. 
Docker installation is failing after certain time on macOS. This job is failing in all the PRs raised in devstack repository. We are currently investigating the issue. 
**Related thread** : https://twou.slack.com/archives/C04ACDVM6A1/p1729108095029319

### CODE REVIEW

**Review by UsamaSadiq — APPROVED** [review]
Looks good to merge. Let's get a second approval on this before merging.

**Review by dianakhuang — APPROVED** [review]
looks good!

**Pull request merged**
